EasyLM/models/llama/llama_model.py

```diff
@@ -918,14 +918,14 @@ def __call__(
 
         batch_size, sequence_length = input_ids.shape
 
+        if attention_mask is None:
+            attention_mask = jnp.ones((batch_size, sequence_length))
+
         if position_ids is None:
             if past_key_values is not None:
                 raise ValueError("Make sure to provide `position_ids` when passing `past_key_values`.")
 
-            position_ids = jnp.broadcast_to(jnp.arange(sequence_length)[None, :], (batch_size, sequence_length))
-
-        if attention_mask is None:
-            attention_mask = jnp.ones((batch_size, sequence_length))
+            position_ids = jnp.broadcast_to(jnp.clip(jnp.cumsum(attention_mask, axis=-1) - 1, a_min=0), (batch_size, sequence_length))
 
```

Comment:
The position_ids was not set correctly if the sequences are not left-aligned. Fixed it here.